api/types/second_factor.go

```diff
@@ -29,9 +29,10 @@ import (
 func legacySecondFactorFromSecondFactors(secondFactors []SecondFactorType) constants.SecondFactorType {
 	hasOTP := slices.Contains(secondFactors, SecondFactorType_SECOND_FACTOR_TYPE_OTP)
 	hasWebAuthn := slices.Contains(secondFactors, SecondFactorType_SECOND_FACTOR_TYPE_WEBAUTHN)
+	hasSSO := slices.Contains(secondFactors, SecondFactorType_SECOND_FACTOR_TYPE_SSO)
 
 	switch {
-	case hasOTP && hasWebAuthn:
+	case (hasOTP && hasWebAuthn) || hasSSO:
```

Comment:
This is an interesting semantical change. Up to this point SecondFactorOn meant that all configs had to be set, but now we could have only SSO second factor but have to call it "on" (because we lack a value to express it).

I think this is (mostly?) fine (we typically check for "otp available" or "webauthn config exists" as well), but we should update godocs on SecondFactorOn to mention the new corner-case.

We should also test this scenario in practice when we get to it. Maybe leave a TODO here as well?

Reply:
The only purpose of this is here - https://github.com/gravitational/teleport/pull/47426/commits/82eff9a725ead23ce6ec50cd67ff6d471499da22

If you'd like I can hold off on merging this PR until that one is also ready and approved

Reply:
I wanted to set this to "on" since we are relying on the logic in some places. However this will be unnecessary as soon as https://github.com/gravitational/teleport/pull/47426/commits/82eff9a725ead23ce6ec50cd67ff6d471499da22 is merged, so I'd rather not overcomplicate things. If you'd like I can hold off on merging this until that PR is ready and approved.

Reply:
I'm happy knowing that there is a follow up. I'd rather get this merged than wait.